openedx_cas/backends.py

```diff
@@ -2,9 +2,63 @@
 implementation.
 """
 from social_core.backends.base import BaseAuth
+from social_core.exceptions import AuthFailed
+from django.conf import settings
+from django_cas_ng.backends import CASBackend
+import urllib.parse
+from django_cas_ng.utils import get_cas_client
+from django.contrib.auth import get_user_model
 
+import logging
 
-class CASAuth(BaseAuth):
+UserModel = get_user_model()
+
+logger = logging.getLogger(__name__)
+
+
+class CASBackendOverride(CASBackend):
+
+    def cas_validation(self, request, ticket, service):
+        client = get_cas_client(service_url=service, request=request)
+        username, attributes, pgtiou = client.verify_ticket(ticket)
+
+        if not username:
+            message = f"Couldn't find username associated with ticket{ticket}"
+            logger.error(message)
+            raise AuthFailed(message)
+
+        if attributes and request:
+            request.session['attributes'] = attributes
+
+        if settings.CAS_USERNAME_ATTRIBUTE != 'cas:user' and settings.CAS_VERSION != 'CAS_2_SAML_1_0':
```

Comment:
```suggestion
        if getattr(settings, "CAS_USERNAME_ATTRIBUTE", "cas:user") != 'cas:user' and settings.CAS_VERSION != 'CAS_2_SAML_1_0':
```
This will avoid errors when the setting is not defined